tests/suites/test_suite_ssl.function

```diff
@@ -339,6 +341,230 @@ int mbedtls_mock_tcp_recv_nb( void *ctx, unsigned char *buf, size_t len )
     return mbedtls_test_buffer_get( socket->input, buf, len );
 }
 
+
+/*
+ * Structure with endpoint certificates for SSL comunication tests.
+ */
+typedef struct mbedtls_endpoint_certificate{
+    mbedtls_x509_crt ca_cert;
+    mbedtls_x509_crt cert;
+    mbedtls_x509_crt cert2;
+    mbedtls_pk_context pkey;
+    mbedtls_pk_context pkey2;
+} mbedtls_endpoint_certificate;
+
+/*
+ * Endpoint structure for SSL comunication tests.
+ */
+typedef struct mbedtls_endpoint{
+        const char *name;
+        mbedtls_ssl_context ssl;
+        mbedtls_ssl_config conf;
+        mbedtls_ctr_drbg_context ctr_drbg;
+        mbedtls_entropy_context entropy;
+        mbedtls_mock_socket socket;
+        mbedtls_endpoint_certificate *cert;
+} mbedtls_endpoint;
+
+#define EP( VAR )   ( &( ep -> VAR ) )
+#define CERT( VAR ) ( &( ep_cert -> VAR ) )
+
+/*
+ * Initialises \p ep structure.
+ * \p endpoint_type mast be set as MBEDTLS_SSL_IS_SERVER or
+ * MBEDTLS_SSL_IS_CLIENT.
+ *
+ * \retval  0 on success or other value on error
```

Comment:
"other value on error" doesn't really seem helpful.

Reply:
changed to "0 on success, otherwise error code"